# "Voices Heard" Data Analysis

## I. Imports

In [42]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import nltk
import yake

from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import classification_report, confusion_matrix

In [43]:
df = pd.read_csv("data/voicesHeard.csv")
df = df.drop(labels=["Reports", "Count"], axis=1)
df

Genre  Age   Gpa  Year Gender
0     Academic Support and Resources   27  2.18     2      M
1     Academic Support and Resources   23  3.11     2      F
2     Academic Support and Resources   20  3.68     2      F
3     Academic Support and Resources   20  1.30     2      F
4     Academic Support and Resources   26  2.50     2      F
...                              ...  ...   ...   ...    ...
1000               Food and Cantines   18  1.16     4      M
1001               Food and Cantines   20  4.05     2      M
1002               Food and Cantines   22  1.86     3      M
1003               Food and Cantines   20  2.06     2      M
1004               Food and Cantines   21  2.58     1      M

[1005 rows x 5 columns]

In [51]:
df["Gender"] = df["Gender"].apply(lambda x: 1 if x=="M" else 0)
df["Genre"].astype("category")

TypeError: NDFrame.astype() got an unexpected keyword argument 'inplace'

In [41]:
df1["Count"].unique().tolist()

[1]

#### Variable Breakdown
We have 1,005 entries along 7 feature columns.
Continuous: ```Age```, ```Gpa```
Categorical: ```Genre```, ```Year```, ```Gender```
Textual: ```Reports```
Unknown: ```Count```

Genre  \
0     Academic Support and Resources   
1     Academic Support and Resources   
2     Academic Support and Resources   
3     Academic Support and Resources   
4     Academic Support and Resources   
...                              ...   
1000               Food and Cantines   
1001               Food and Cantines   
1002               Food and Cantines   
1003               Food and Cantines   
1004               Food and Cantines   

                                                Reports  Age   Gpa  Year  \
0     The limited access to research databases and m...   27  2.18     2   
1     I'm having trouble finding the course material...   23  3.11     2   
2     It's frustrating to have limited access to res...   20  3.68     2   
3     I'm really struggling in one of my classes but...   20  1.30     2   
4      I am really struggling with understanding the...   26  2.50     2   
...                                                 ...  ...   ...   ...   
1000  26. I have been unable to find food that meets...   18  1.16     4   
1001  27. I have been unable to find food that I can...   20  4.05     2   
1002  28. I have been unable to find food that I enjoy.   22  1.86     3   
1003  29. I have been unable to find food that is he...   20  2.06     2   
1004  30. I have been unable to find food that is co...   21  2.58     1   

      Count  isMale  
0         1       1  
1         1       0  
2         1       0  
3         1       0  
4         1       0  
...     ...     ...  
1000      1       1  
1001      1       1  
1002      1       1  
1003      1       1  
1004      1       1  

[1005 rows x 7 columns]

In [38]:
df1["Genre"] = df1["Genre"].astype("category")
df1["Gender"] = df1["Gender"].astype("category")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1005 entries, 0 to 1004
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype   
---  ------   --------------  -----   
 0   Genre    1005 non-null   category
 1   Reports  1005 non-null   object  
 2   Age      1005 non-null   int64   
 3   Gpa      1005 non-null   float64 
 4   Year     1005 non-null   int64   
 5   Count    1005 non-null   int64   
 6   isMale   1005 non-null   int64   
dtypes: category(1), float64(1), int64(4), object(1)
memory usage: 48.6+ KB
